docs_src/design/adr/0019-Service-List.md

```diff
@@ -0,0 +1,116 @@
+
+### Considerations
+
+- In some cases, the list of services is needed before the services are up (security needs to know so appropriate tokens can be generated and handed out when the services bootstrap).
+- In some cases, a service may be “disabled” at runtime because of resource constraints (example: notification service is turned off as it is not used and there is a need to operate with limited memory and CPU capability).  Security would still need to know about disabled services in order to create a token for such services that may be enabled later.  However, the other services (or 3rd party applications) may need to know that they are disabled vs enabled.
+- In the case that the services are distributed (especially when device services run on different hosts - as is likely), access to a distributed file system or volume mount becomes more of an issue and requires additional technology.  *Note: distributing device services while in secure mode is now more complex due to the fact that all services require Vault access and the services get their Vault token from file.  Distributed device services in secure mode would therefore would require a distributed file system.*
+- The service list would have to be a list of services and service locations (URI).  Therefore, the service list will really be a map of service names (keys) and the location of the services URIs (values) and potentially other information such as `known secrets` as used by the secret store setup (`redisdb` is the only `known secret` to date).  This will help support things like the API gateway setup as well.  In some cases, just a list of the service names is needed (in security token setup for example).  In this latter case, the service would just pick off the service keys.
+- Generation of security tokens for the new services under most of the conceived alternatives is complex.  In the current security bootstrapping, secret store setup will require a restart of all the services in order to add or remove a service.  The ORRA project and some adopters are clear that they would like to have the ability to add/remove services dynamically at runtime.  *Note: in some circumstances, services can be added/removed dynamically today, but not when EdgeX security services are enabled.  CLI and UI also do not allow for the addition/remove of services - such as device or application services.*
```

Comment:
This is now out of date with recent enhancements. Secret Store setup can now be restarted w/o requiring the other service to be restarted.